diff --git a/third_party/farmhash/workspace.bzl b/third_party/farmhash/workspace.bzl
index f273313891d..421a582bf76 100644
--- a/third_party/farmhash/workspace.bzl
+++ b/third_party/farmhash/workspace.bzl
@@ -7,8 +7,8 @@ def repo():
 
     # Attention: tools parse and update these lines.
     # LINT.IfChange
-    FARMHASH_COMMIT = "0d859a811870d10f53a594927d0d0b97573ad06d"
-    FARMHASH_SHA256 = "18392cf0736e1d62ecbb8d695c31496b6507859e8c75541d7ad0ba092dc52115"
+    FARMHASH_COMMIT = "be24c1508eacdf25691bcf561b6f443d6e3ed141"
+    FARMHASH_SHA256 = "965aa1fa293fd1d934305d2c4fd6472f817a55f33f66ae5ed300496d8845d452"
     # LINT.ThenChange(//tensorflow/lite/tools/cmake/modules/farmhash.cmake)
 
     tf_http_archive(
@@ -16,7 +16,7 @@ def repo():
         build_file = "//third_party/farmhash:farmhash.BUILD",
         sha256 = FARMHASH_SHA256,
         strip_prefix = "farmhash-{commit}".format(commit = FARMHASH_COMMIT),
-        urls = tf_mirror_urls("https://github.com/google/farmhash/archive/{commit}.tar.gz".format(commit = FARMHASH_COMMIT)),
+        urls = tf_mirror_urls("https://github.com/qnx-ports/farmhash/archive/{commit}.tar.gz".format(commit = FARMHASH_COMMIT)),
     )
 
     tf_http_archive(
@@ -25,5 +25,5 @@ def repo():
         patch_file = ["//third_party/farmhash:farmhash_support_cuda.patch"],
         sha256 = FARMHASH_SHA256,
         strip_prefix = "farmhash-{commit}".format(commit = FARMHASH_COMMIT),
-        urls = tf_mirror_urls("https://github.com/google/farmhash/archive/{commit}.tar.gz".format(commit = FARMHASH_COMMIT)),
+        urls = tf_mirror_urls("https://github.com/qnx-ports/farmhash/archive/{commit}.tar.gz".format(commit = FARMHASH_COMMIT)),
     )
diff --git a/third_party/xla/third_party/tsl/third_party/farmhash/workspace.bzl b/third_party/xla/third_party/tsl/third_party/farmhash/workspace.bzl
index f273313891d..421a582bf76 100644
--- a/third_party/xla/third_party/tsl/third_party/farmhash/workspace.bzl
+++ b/third_party/xla/third_party/tsl/third_party/farmhash/workspace.bzl
@@ -7,8 +7,8 @@ def repo():
 
     # Attention: tools parse and update these lines.
     # LINT.IfChange
-    FARMHASH_COMMIT = "0d859a811870d10f53a594927d0d0b97573ad06d"
-    FARMHASH_SHA256 = "18392cf0736e1d62ecbb8d695c31496b6507859e8c75541d7ad0ba092dc52115"
+    FARMHASH_COMMIT = "be24c1508eacdf25691bcf561b6f443d6e3ed141"
+    FARMHASH_SHA256 = "965aa1fa293fd1d934305d2c4fd6472f817a55f33f66ae5ed300496d8845d452"
     # LINT.ThenChange(//tensorflow/lite/tools/cmake/modules/farmhash.cmake)
 
     tf_http_archive(
@@ -16,7 +16,7 @@ def repo():
         build_file = "//third_party/farmhash:farmhash.BUILD",
         sha256 = FARMHASH_SHA256,
         strip_prefix = "farmhash-{commit}".format(commit = FARMHASH_COMMIT),
-        urls = tf_mirror_urls("https://github.com/google/farmhash/archive/{commit}.tar.gz".format(commit = FARMHASH_COMMIT)),
+        urls = tf_mirror_urls("https://github.com/qnx-ports/farmhash/archive/{commit}.tar.gz".format(commit = FARMHASH_COMMIT)),
     )
 
     tf_http_archive(
@@ -25,5 +25,5 @@ def repo():
         patch_file = ["//third_party/farmhash:farmhash_support_cuda.patch"],
         sha256 = FARMHASH_SHA256,
         strip_prefix = "farmhash-{commit}".format(commit = FARMHASH_COMMIT),
-        urls = tf_mirror_urls("https://github.com/google/farmhash/archive/{commit}.tar.gz".format(commit = FARMHASH_COMMIT)),
+        urls = tf_mirror_urls("https://github.com/qnx-ports/farmhash/archive/{commit}.tar.gz".format(commit = FARMHASH_COMMIT)),
     )
